### DISCUSSION TITLE
"provider-config.yaml" Configuration Not Reflecting in Source List

### DISCUSSION THREAD

**Discussion by iis2h** [projectdiscovery/subfinder#1189] [Q&A] [answered]
When I ran `subfinder -ls`, it displayed the path of the providers file, which is:

`[INF] You can modify /Users/<USER>/.config/subfinder/provider-config.yaml to configure your keys/tokens.`

So, I modified the file and added the API keys as follows:

```bash
binaryedge: 
  - <API_KEY>
bufferover: []
builtwith: []
c99: []
censys:
  - <API_ID>:<API_Secret>
securitytrails:
  - <API_KEY>
shodan:
  - <API_KEY>
 ```
As you can see, I added API keys for `binaryedge` , `censys` , `securitytrails` , and `shodan`. Then, I ran `subfinder -ls` and the result was:
```bash
[INF] Current subfinder version v2.6.5 (latest)
[INF] Current list of available sources. [39]
[INF] Sources marked with an * need key(s) or token(s) to work.
[INF] You can modify /Users/<USER>/.config/subfinder/provider-config.yaml to configure your keys/tokens.

alienvault
anubis
bevigil *
binaryedge *
bufferover *
c99 *
censys *
.
.
.
securitytrails *
shodan *
waybackarchive
```
When I ran `subfinder -pc /Users/<USER>/.config/subfinder/provider-config.yaml -ls`, the same result returned

**Comment by olearycrew** [comment, marked as answer]
I believe `-ls` just lists what sources are possible and which would need an API key. I don't think it does any API verification to those keys, though that might be a good idea.

e.g. I think that the output of `-ls` is unchanged by the provider-config

**Reply by iis2h** [reply]
Oh, I think when I add an API key, the asterisk mark will be removed when I execute `subfinder -ls`. I remember there is a tool that confirms the API was added by removing the asterisk mark, but I can't remember which one it is. Anyway, thanks for your great work
